thekuuzzz/PrefParser.py

```diff
@@ -0,0 +1,139 @@
+
+    def _findIn(searchSpace, item):
+        # Recursive search for items inside a nested dictionary
+        foundItem = []
+        try:
+            for c, d in searchSpace.items():
+                if c == item:
+                    foundItem.append(d)
+                else:
+                    if isinstance(d, dict):
+                        nestResult = findIn(d, item)
+                        for result in nestResult:
+                            foundItem.append(result)
+                        else:
+                            if isinstance(d, dict):
+                                for nestedItems in d:
+                                    if isinstance(nestedItems, dict):
+                                        extraItems = findIn(nestedItems, item)
+                                        for evenMoreItems in extraItems:
+                                            foundItem.append(extraItems)
+            return foundItem
+        except Exception as e:
+            print("Exception running application, here's what we know")
+            print(e)
+            return 0
+
+    def _dateConversion(webkitTimestamp):
+        # Converts date from webkit time to ISO 8601 time
+        startDate = datetime.datetime(1601, 1, 1, 0, 0, 0, 0)
+        currentDate = datetime.timedelta(microseconds=int(webkitTimestamp))
+        isoTime = startDate + currentDate
+        return isoTime
+
+    def parseName():
+        # Parse through and get the account name, email, and profile name
+        # Returns the name, email and profile name
+        try:
+
+            for d in json_object['account_info']:
+                name = d.get('full_name')
+                email = d.get('email')
+                for c in d['profile']:
+                    profile = c.get('name')
+
+                    return name, email, profile
+        except Exception as e:
+            print('Exception caught: ' + e)
+            raise EnvironmentError.filename
+            sys.exit
+
+    def parseDate():
+        # Parses the last time chrome was updated in ISO 8601 UTC format
+        # Returns the last update date in ISO 8601 UTC
+        try:
+
+            webKitTime = json_object['last_update_date']
+            finalDate = _dateConversion(webKitTime)
+            return finalDate
+        except Exception as e:
+            print("Exception running application, here's what we know")
+            print(e)
+            return 0
+
+    def parseMedia():
+        # Parses the following information from the media_engagement key
+        # 1: List of Websites 2: last modified 3. Media played? 4. visitcount
+        # returns these keys in a dict
+        try:
+            media = {}
+            for c, d in json_object['profile']['content_settings']['exceptions']['media_engagement'].iteritems():
+                media.append(prefs.items(website=c, lastModified=dateConversion(d['last_modified']), playback=mediaPlaybacks, visitCount=visits))
```

Comment:
what is prefs.items? I don't see that anywhere.... don't tell me that's another global variable?